File: cpp/src/_70.cpp
#include <bits/stdc++.h>
using namespace std;

/*
class Solution {
public:
    int climbStairs(int n) {
        vector<int> dp(n+1,0);
        dp[1] = 1;
        dp[0] = 1;
        for(int i = 2;i<n+1;++i)
        {
            dp[i] = dp[i-1] + dp[i-2];
        }
        return dp[n];
    }
};
*/
class Solution {
public:
    int climbStairs(int n) {
        
        int dp_1 = 1;
        int dp_0 = 1;
        int dp;
        while(--n)
        {
            dp = dp_1 + dp_0;
            dp_0 = dp_1;
            dp_1 = dp;
        }
        return dp;
    }
};

int main()
{
    return 0;
}

File: 70.py
class Solution:
    def climbStairs(self, n: int) -> int:
        f0 = 1
        f1 = 1
        f2 = f0 + f1
        for _ in range(n-2):
            f0, f1 = f1, f2
            f2 = f0 + f1
        return f2 if n > 1 else 1
